edx_django_utils/monitoring/middleware.py

```diff
@@ -63,13 +63,15 @@ def accumulate_metric(cls, name, value):
         metrics_cache.set(name, value)
 
     @classmethod
-    def _batch_report(cls):
+    def _batch_report(cls, request):
         """
         Report the collected custom metrics to New Relic.
         """
         if not newrelic:
             return
         metrics_cache = cls._get_metrics_cache()
+        if request.user.id:
```

Comment:
I would also check request.user, since I think that will be None if the user isn't logged in... something like:

```python
if getattr(request, 'user') is not None and request.user.id:
    newrelic.agent.add_custom_parameter('user_id', request.user.id)
```